File: SYP.Web/Modules/ServerTypes/Order/OrderDetailColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderDetailRow } from "./OrderDetailRow";

export interface OrderDetailColumns {
    Id: Column<OrderDetailRow>;
    OrderNumber: Column<OrderDetailRow>;
    ProductCodeName: Column<OrderDetailRow>;
    Quantity: Column<OrderDetailRow>;
    UnitCode: Column<OrderDetailRow>;
    UnitPrice: Column<OrderDetailRow>;
    VatRateName: Column<OrderDetailRow>;
    VatRate: Column<OrderDetailRow>;
    Discount: Column<OrderDetailRow>;
    LineTotal: Column<OrderDetailRow>;
    Notes: Column<OrderDetailRow>;
}

export class OrderDetailColumns extends ColumnsBase<OrderDetailRow> {
    static readonly columnsKey = 'Order.OrderDetail';
    static readonly Fields = fieldsProxy<OrderDetailColumns>();
}

File: SYP.Web/Modules/Order/OrderDetail/OrderDetailColumns.cs
namespace SYP.Order.Columns;

[ColumnsScript("Order.OrderDetail")]
[BasedOnRow(typeof(OrderDetailRow), CheckNames = true)]
public class OrderDetailColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string OrderNumber { get; set; }
    public string ProductCodeName { get; set; }
    public decimal Quantity { get; set; }
    public string UnitCode { get; set; }
    public decimal UnitPrice { get; set; }
    public string VatRateName { get; set; }
    public decimal VatRate { get; set; }
    public decimal Discount { get; set; }
    public decimal LineTotal { get; set; }
    [EditLink]
    public string Notes { get; set; }
}